diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index bc2fe47..452531e 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -9153,10 +9153,10 @@ static void check_preempt_wakeup_fair(struct rq *rq, struct task_struct *p, int
 	/*
 	 * Likewise, if @p is heavier than current (under BORE: more
 	 * interactive), per kernel.sched_burst_protect_slice_lv, override
 	 * current's slice protection to allow preemption.
 	 */
 	if (do_preempt_weight(cfs_rq, pse, se))
-		do_preempt_short = true;
+		preempt_action = PREEMPT_WAKEUP_SHORT;
 #endif
 
 	/*
@@ -14329,2 +14329,15 @@ __init void init_sched_fair_class(void)
 #endif
-}
+}
+
+#ifdef CONFIG_SCHED_BORE
+void reweight_task(struct task_struct *p, int prio)
+{
+	struct sched_entity *se = &p->se;
+	struct cfs_rq *cfs_rq = cfs_rq_of(se);
+	struct load_weight *load = &se->load;
+	unsigned long weight = scale_load(sched_prio_to_weight[prio]);
+
+	reweight_entity(cfs_rq, se, weight);
+	load->inv_weight = sched_prio_to_wmult[prio];
+}
+#endif /* CONFIG_SCHED_BORE */
